### ISSUE TITLE
False positives for required single space after/before parens

### ISSUE THREAD

**Issue by bounasserabdelwahab** [squizlabs/PHP_CodeSniffer#1620]
Now I'm not really sure if this is a problem with PHPCS or with [WordPress-core coding standard](https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards) that I'm currently using, but I found that **phpcbf & phpcs do not detect & fix required single spaces after and before function parenthesis** when function parameters are on multiple lines, here's what I mean:

```php
<?php

// Required single space before and after, this error is detected by phpcs.
english_to_french(array( 'one' => 'un', 'two' => 'deux', 'three' => 'trois' ));

// False positive, required single spaces are not detected.
english_to_french(array(
  'one' => 'un',
  'two' => 'deux',
  'three' => 'trois',
));

// Another false positive.
fart(function () {
  // Do stuff.
});

// No errors, as expected.
english_to_french( array(
  'one' => 'un',
  'two' => 'deux',
  'three' => 'trois',
) );

// PHPCS doesn't care about how many spaces, this is obviously a false positive.
english_to_french(          array(  // lots of spaces.
  'one' => 'un',
  'two' => 'deux',
  'three' => 'trois',
)             ); // lots of spaces.
```

I hope I explained my point appropriately. =)

**Comment by jrfnl** [comment]
@bounasserabdelwahab Please report this with the WordPress Coding Standards project and we'll have a look.

**Comment by jrfnl** [comment]
@bounasserabdelwahab I went ahead and took a look anyway.

The problem is that your code is multi-line, but that the formatting is not a "proper" multi-line function where each parameter starts on a new line.
```php
proper_multi_line(
   $param1,
   $param2,
);
```

The WordPress Coding Standards use the PHPCS `PEAR.Functions.FunctionCallSignature` sniff to check the formatting for this type of code.
But as the WordPress Coding Standards allow for "mixed" single/multi-line function calls, you don't see the open/close parenthesis whitespace errors for those mixed multi-line function calls.

However, these issues would be correctly fixed when using `phpcbf` as the two errorcodes which are silenced by WPCS to allow for "mixed" line function calls, are **not** silenced for the fixer.
In effect, this means that the fixer will change the function call to be a "proper" multi-line call and will by doing so, fix the whitespace.

You'll need to make sure you use WPCS 0.12.0 or higher for this. Older versions of WPCS silenced that part for the fixer as well.

I'd still suggest opening an issue with WPCS to see if something can be done so that the "too little/too much whitespace around parenthesis" for multi-line mixed function calls will be reported too.

**Comment by bounasserabdelwahab** [comment]
@jrfnl Thank you for your response! Updating to the latest version of `WPCS` barely fixed the issue and we're almost there.

```PHP
// Original.
fun(array(
  'one'   => 'un',
  'two'   => 'deux',
  'three' => 'trois',
), 'outside');

// Fixed.
fun(
  array(
    'one'   => 'un',
    'two'   => 'deux',
    'three' => 'trois',
  ), 'outside'
);
```
As you might tell, the only problem that is still present is that the fixer wraps the paramters in a new line and indents the params, instead of adding a single space. The code above should be fixed to this:

```PHP
// Fixed.
fun( array(
  'one'   => 'un',
  'two'   => 'deux',
  'three' => 'trois',
), 'outside' );
```

**Comment by jrfnl** [comment]
@bounasserabdelwahab This has been discussed in detail with representatives of the WP Core team and it was decided that the fix as is being made now is ok.

**Comment by bounasserabdelwahab** [comment]
@jrfnl If that's true then that would mean that `php-cs-fixer` is useless for my needs for now. I guess I'll stick to using `phpcs` alone. Thank you for your responses, I'll close this for now =)

**Comment by jrfnl** [comment]
@bounasserabdelwahab You could choose to exclude that particular sniff completely for the fixer, but still use the fixers from the other sniffs. Maybe that helps a little ?
